modules/repository-url/src/test/java/org/elasticsearch/common/blobstore/url/HttpURLBlobStoreTests.java

```diff
@@ -126,8 +127,14 @@ String getBlobName() {
 
     public void testRangeReadOutsideOfLegalRange() {
         BlobContainer container = getBlobContainer();
-        expectThrows(IllegalArgumentException.class, () -> container.readBlob(blobName, -1, content.length).read());
-        expectThrows(IOException.class, () -> container.readBlob(blobName, content.length + 1, content.length).read());
+        expectThrows(
+            IllegalArgumentException.class,
+            () -> container.readBlob(OperationPurpose.SNAPSHOT, blobName, -1, content.length).read()
```

Comment:
It would be nice to randomize the purpose here (and in many places elsewhere). I think it can be a follow-up though.

Reply:
Yep we should have follow-ups for actual usages of the purpose parameter. In this PR, all callsite simply uses the SNAPSHOT as the default.